src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_fankriss.cpp

```diff
@@ -41,155 +42,115 @@ class boss_fankriss : public CreatureScript
-                                if (Hatchling)
-                                    Hatchling->AI()->AttackStart(target);
-                                Hatchling = me->SummonCreature(15962, target->GetPositionX()-5, target->GetPositionY()-5, target->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000);
-                                if (Hatchling)
-                                    Hatchling->AI()->AttackStart(target);
-                                Hatchling = me->SummonCreature(15962, target->GetPositionX()-5, target->GetPositionY()+5, target->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000);
-                                if (Hatchling)
-                                    Hatchling->AI()->AttackStart(target);
-                                break;
+                        case 0:
+                            DoTeleportPlayer(pUnit, -8106.0142f, 1289.2900f, -74.419533f, 5.112f);
+                            break;
+                        case 1:
+                            DoTeleportPlayer(pUnit, -7990.135354f, 1155.1907f, -78.849319f, 2.608f);
+                            break;
+                        case 2:
+                            DoTeleportPlayer(pUnit, -8159.7753f, 1127.9064f, -76.868660f, 0.675f);
+                        }
+                        if (Creature* pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 3, pUnit->GetPositionY() - 3, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
+                        {
+                            summoned.Summon(pCreature);
+                            pCreature->AI()->AttackStart(pUnit);
+                        }
+                        if (Creature* pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 3, pUnit->GetPositionY() + 3, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
+                        {
+                            summoned.Summon(pCreature);
+                            pCreature->AI()->AttackStart(pUnit);
+                        }
+                        if (Creature* pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 5, pUnit->GetPositionY() - 5, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
+                        {
+                            summoned.Summon(pCreature);
+                            pCreature->AI()->AttackStart(pUnit);
+                        }
+                        if (Creature* pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 5, pUnit->GetPositionY() + 5, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000))
+                        {
+                            summoned.Summon(pCreature);
+                            pCreature->AI()->AttackStart(pUnit);
                         }
+                        break;
```

Comment:
The `pCreature` will be only one? or there is a case where you will `summon` 4 creatures?
if not, you can write this more short  like this way I think:
```C++
Creature* pCreature;

if (
  pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 3, pUnit->GetPositionY() - 3, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000) || 
  pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 3, pUnit->GetPositionY() + 3, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000) ||
  pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 5, pUnit->GetPositionY() - 5, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000) ||
  pCreature = me->SummonCreature(15962, pUnit->GetPositionX() - 5, pUnit->GetPositionY() + 5, pUnit->GetPositionZ(), 0, TEMPSUMMON_TIMED_DESPAWN_OUT_OF_COMBAT, 15000)
) {
    summoned.Summon(pCreature);
    pCreature->AI()->AttackStart(pUnit);
}
```

but I am not sure if one of these if exclude the others, if they were several `if / else if` I would be sure about this suggestion